src/Type/Doctrine/Query/QueryResultTypeWalker.php

```diff
@@ -1097,8 +1127,12 @@ public function walkLiteral($literal)
 				break;
 
 			case AST\Literal::BOOLEAN:
-				$value = strtolower($literal->value) === 'true' ? 1 : 0;
-				$type = new ConstantIntegerType($value);
```

Comment:
This was just wrong for postgres, that returns real booleans, but we were infering `1|0|'1'|'0'`